Add Upload and Download Commands

U8NWXD opened the pull request:
Requires `vivarium-core` commit `c4a22331c294067f20bdff8fd226818d5f521b15`, which is merged in vivarium-collective/vivarium-core#16. This PR should not be merged until the vivarium-core PR is merged.

We download experiments with `python access_db.py download <experiment_id>` and get a JSON file like 
[this](https://github.com/vivarium-collective/vivarium-scripts/files/5302121/20200921.135657.json.zip). We can then upload such a file with `python access_db.py upload <path_to_json>`. Note that experiment IDs are preserved, so the destination database must not already have an experiment with an ID that matches the one being uploaded.

Here are analysis plots of an experiment before and after it was downloaded and re-uploaded:

Before:

![agents](https://user-images.githubusercontent.com/19878639/94622518-5e886a80-0280-11eb-8a39-2b0fab0793d0.png)

After:

![agents](https://user-images.githubusercontent.com/19878639/94622534-647e4b80-0280-11eb-8e68-8bd09ae6a8c4.png)

U8NWXD committed:
Add Upload and Download Commands

Requires `vivarium-core` commit
c4a22331c294067f20bdff8fd226818d5f521b15.

eagmon reviewed (approved):
Very nice!

1) You'll need to add vivarium-core as a dependency in setup
2) What happens if the experiment_id is already in the db? Since duplicate experiment_ids are probably rare, it would be ok to replace the experiment since it is probably the same experiment in question. But we should not make duplicate entries of each data entry.

eagmon commented on scripts/access_db.py:
```diff
@@ -81,6 +95,30 @@ def delete(self, args):
+                downloaded = json.load(f)
+            data = downloaded['data']
+            environment_config = downloaded['environment_config']
+            experiment_id = downloaded['experiment_id']
```
How would the ```experiment_id``` get into the db with this? I don't see it being passed into ```data_to_database```

U8NWXD replied:
Thanks for pointing this out. We don't actually use the experiment ID that shows up in the code. The ID is already part of the downloaded experiment data because each document has an `experiment_id` key:

![Screen Shot 2020-09-30 at 09 54 40](https://user-images.githubusercontent.com/19878639/94694633-0d6d8a80-0303-11eb-86cd-5c62dee9e0e8.png)

This means we don't really need to handle the experiment ID separately. I'll fix that

U8NWXD replied:
Done in 2258132725ecc15fa9e30be448feab90ec4c8a42

eagmon replied:
@U8NWXD -- We do use the ```experiment_id``` to query the DB. If it is not included in this upload, I'm not sure we'll be able to find the experiment. For example, we use ```experiment_id``` in ```vivarium-scripts```.

U8NWXD replied:
@eagmon I'm not sure I understand. Yes, we need the experiment ID, but the ID is already in every object we download from the database, so we don't need to store a separate copy here. More concretely, the experiment ID is already in `environment_config` and `data` in the code this comment references.

eagmon replied:
Ok, I see now. Thanks @U8NWXD!

U8NWXD commented:
Regarding (1): `vivarium-core` is already a dependency in setup. I added it in my CLI refactor PR:

https://github.com/vivarium-collective/vivarium-scripts/blob/57834ff9d5d25104114dd11d40a409930176fc36/setup.py#L30

Regarding (2): Right now, it will probably introduce duplicate documents, which is bad! I'll add a check to make sure the ID isn't already present.

**Edit:** Done in ffcdd1c5cf29e341e68a03835f93a8d09b2bab00

U8NWXD committed:
Stop handling experiment_id separately

experiment_id is already in all the downloaded documents, so there's no
need to include it again.

U8NWXD committed:
Abort before uploading duplicate experiment
